src/main/java/org/opensearch/flowframework/FlowFrameworkPlugin.java

```diff
@@ -51,7 +53,10 @@ public Collection<Object> createComponents(
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<RepositoriesService> repositoriesServiceSupplier
     ) {
-        WorkflowStepFactory workflowStepFactory = new WorkflowStepFactory(client);
+        Settings settings = environment.settings();
+        // TODO: Creating NodeClient is a temporary fix until we get the NodeClient from the provision API
+        NodeClient nodeClient = new NodeClient(settings, threadPool);
+        WorkflowStepFactory workflowStepFactory = new WorkflowStepFactory(client, nodeClient);
```

Comment:
It's of type `Client` [here](https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/52/files#diff-9d60d6086ce87e2240d7d6bcf8d2a8b79c73400fc6ec6865f74fc571e03bf1afR44). Typecasting it to `(NodeClient) client` fails the test. Anyway, it's temporary until we get the NodeClient from provision API.

Reply:
> It's of type `Client` [here](https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/52/files#diff-9d60d6086ce87e2240d7d6bcf8d2a8b79c73400fc6ec6865f74fc571e03bf1afR44). 

And that is called from Node.java passing this object: `private final NodeClient client;`.

> Typecasting it to `(NodeClient) client` fails the plugin test. 

Can't you typecast the param you're sending to the `WorkflowStepFactory` constructor?

> Anyway, it's temporary until we get the NodeClient from provision API.

Not clear how this is relevant.  When this Plugin is loaded in OpenSearch, `client` will refer to the `NodeClient` instance initialized on the `Node`.

Reply:
> Can't you typecast the param you're sending to the `WorkflowStepFactory` constructor?
I can. The tests are failing with the below error
```
org.opensearch.flowframework.workflow.WorkflowProcessSorterTests > classMethod FAILED
    java.lang.ClassCastException: class org.opensearch.client.Client$MockitoMock$zUodJfPz cannot be cast to class org.opensearch.client.node.NodeClient (org.opensearch.client.Client$MockitoMock$zUodJfPz and org.opensearch.client.node.NodeClient are in unnamed module of loader 'app')
        at __randomizedtesting.SeedInfo.seed([6CC035B077521807]:0)
        at org.opensearch.flowframework.workflow.WorkflowStepFactory.populateMap(WorkflowStepFactory.java:43)
        at org.opensearch.flowframework.workflow.WorkflowStepFactory.<init>(WorkflowStepFactory.java:37)
        at org.opensearch.flowframework.workflow.WorkflowProcessSorterTests.setup(WorkflowProcessSorterTests.java:67)

Reply:
> java.lang.ClassCastException: class org.opensearch.client.Client$MockitoMock$zUodJfPz cannot be cast to class org.opensearch.client.node.NodeClient 

I think the issue is that you are mocking `Client.class` rather than `NodeClient.class`.  Mockito doesn't seem to allow downcasting (and rightly so).